web/src/main/java/org/cbioportal/web/SampleController.java

```diff
@@ -261,7 +267,12 @@
     
     public List<Sample> fetchSamplesInner(
         List<String> sampleListIds, String projection
-    ) {
+    ) throws SampleListNotFoundException {
+        for (String sampleListId : sampleListIds) {
```

Comment:
Hmm. Not a huge fan of iteratively calling service methods. I'm approving this, but can you just verify that this isn't a performance concern? Maybe just call it with ~ 100 valid sample list IDs.

Reply:
@Luke-Sikina I don't think this will cause a performance issue, the reason is that `fetchSamplesInner` is an internally used function in `SampleController`, and the only use case is like [this](https://github.com/cBioPortal/cbioportal/blob/master/web/src/main/java/org/cbioportal/web/SampleController.java#L246-L248):
```
for (String sampleListId : sampleListIds) {
    samples.addAll(fetchSamplesInner(Arrays.asList(sampleListId), projection.name()));
 }
```
So this `fetchSamplesInner` function will be called multiple times if there are more than one `sampleListIds`, but each time it will only accept a single sample list id.

However this is not very good since the function signature of `fetchSamplesInner` indicate that this function will accept a list of sample list ids. Maybe I can change `fetchSampleInner` to this?
```
public List<Sample> fetchSamplesInner(
    String sampleListId, String projection
) throws SampleListNotFoundException {
    // check sampleList
    // if sampleList not found, throw sampleListNotFound exception
    sampleListService.getSampleList(Arrays.asList(sampleListId));
    return sampleService.fetchSamples(sampleListIds, projection);
}
```